primitives/arithmetic/src/traits.rs

```diff
@@ -65,7 +53,54 @@ impl<T:
 	CheckedShl + CheckedShr + CheckedAdd + CheckedSub + CheckedMul + CheckedDiv +
 	Saturating + PartialOrd<Self> + Ord + Bounded +
 	HasCompact + Sized
-> SimpleArithmetic for T {}
+> BaseArithmetic for T {}
+
+/// A meta trait for arithmetic.
+///
+/// Arithmetic types do all the usual stuff you'd expect numbers to do. They are guaranteed to
+/// be able to represent at least `u32` values without loss, hence the trait implies `From<u32>`
+/// and smaller ints. All other conversions are fallible.
+pub trait RuntimeArithmetic:
+	BaseArithmetic +
+	From<u8> + From<u16> + From<u32> + TryInto<u8> + TryInto<u16> + TryInto<u32> + TryFrom<u64> +
+	TryInto<u64> + TryFrom<u128> + TryInto<u128> + TryFrom<usize> + TryInto<usize> +
+	UniqueSaturatedInto<u8> + UniqueSaturatedInto<u16> + UniqueSaturatedInto<u32> +
+	UniqueSaturatedFrom<u64> + UniqueSaturatedInto<u64> + UniqueSaturatedFrom<u128> +
+	UniqueSaturatedInto<u128>
+{}
+impl<T:
+	BaseArithmetic +
+	From<u8> + From<u16> + From<u32> + TryInto<u8> + TryInto<u16> + TryInto<u32> + TryFrom<u64> +
+	TryInto<u64> + TryFrom<u128> + TryInto<u128> + TryFrom<usize> + TryInto<usize> +
+	UniqueSaturatedInto<u8> + UniqueSaturatedInto<u16> + UniqueSaturatedInto<u32> +
+	UniqueSaturatedFrom<u64> + UniqueSaturatedInto<u64> + UniqueSaturatedFrom<u128> +
+	UniqueSaturatedInto<u128>
+> RuntimeArithmetic for T {}
+
+/// Same as [`RuntimeArithmetic`] for integer types of any size, without any assumption of their size
+/// being at least `u32`.
+pub trait AnyArithmetic:
```

Comment:
Let me see if I get you right: 

What you recommend is putting all the `TryInto<_>`, `TryFrom<_>`, `UniqueSaturatedInto<_>` and `UniqueSaturatedFrom<_>`s in `Base` for all the 6 major integer types, since we don't know the size there, right? and then we can say `AtLeast32Bit: Base + From<u8> + From<u16> + From<u32>`.

The only minor problem with that would be that some of the traits would be duplicate and don't really make sense. I.e. a type ends up being `From<u32> + TryFrom<u32>`. Is this cool?

Reply:
1. Yeah base takes all these trait bounds and I also would give it `From<u8>` (there is no datatype that is smaller than u8)
2. Every type implements `From<Self>` and also 
```
impl<T, U> TryFrom<U> for T where
    U: Into<T>, 
```
   This means that every type automatically implements `TryFrom` for all types it implements `From` for.